enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,320 @@
+In the context of this enhancement, [AWS Load Balancer Operator](/enhancements/ingress/aws-load-balancer-operator.md) is the expected implementation.
+
+### Drawbacks
+
+In the absence of the default ingress controller, the Route API is left unserved. OpenShift components or workloads can create routes, but there is no available controller to support them. An important consequence of this is the impact on the ingress API which relies on [the ingress to route translation](#ingress-to-route-controller). All this may be misleading for the users.
+
+## Design Details
+
+### Open Questions [optional]
+
+N/A
+
+### Test Plan
+TBD
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+N/A
+
+#### Tech Preview -> GA
+N/A
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Same as [the component selection strategy](/enhancements/installer/component-selection.md#upgrade--downgrade-strategy).     
+Note that:
+- the capabilities [cannot disabled post install](/enhancements/installer/component-selection.md#capabilities-cannot-be-uninstalled)
+- the capabilities can be [implicitly enabled](/enhancements/installer/component-selection.md#updates) (e.g. cluster version with `None` baseline upgrades to the version which moves some of the core payloads to a capability)
+
+### Version Skew Strategy
+N/A
+
+### Operational Aspects of API Extensions
+N/A - no api extensions are being introduced in this proposal
+
+#### Failure Modes
+N/A - no api extensions are being introduced in this proposal
+
+#### Support Procedures
+N/A - no api extensions are being introduced in this proposal
+
+## Implementation History
+N/A
+
+## Alternatives
+
+### Implement the ingress capability for the standalone OpenShift
+
+The goal of this enhancement is to satisfy the HyperShift's requirements. The implementation of the ingress capability for the standalone OpenShift brings a lot of additional complexity, unnecessary for HyperShift. Most probably the following components would need to be updated to tolerate the absence of the ingress operator:
+- The openshift installer. It [checks for the admission of the console route](https://github.com/wking/openshift-installer/blob/3d49ffbff57011bbf846bf91de11526bb3543a83/cmd/openshift-install/create.go#L543) by the ingress controller.
+- The authentication operator. It checks the availability of [the authentication server's](https://github.com/openshift/cluster-authentication-operator/blob/da39951a53ad95be28e32a48f278cc23f41e99a7/pkg/controllers/oauthendpoints/oauth_endpoints_controller.go#L30-L31) 
+and [console](https://github.com/openshift/cluster-authentication-operator/blob/da39951a53ad95be28e32a48f278cc23f41e99a7/pkg/controllers/configobservation/console/observe_consoleurl.go#L15) routes.
```

Comment:
It doesn't look like that authentication operator check is looking up the route.  However, the console operator does perform health checks on its route: https://github.com/openshift/console-operator/blob/9fefc7fe89542e94ec4e5bb98b275716df51c311/pkg/console/controllers/healthcheck/controller.go

Reply:
By the way, the authentication operator also expects the ingress operator to create the "router-certs" secret and the "default-ingress-cert" configmap with the default ingress wildcard certificate (see <https://github.com/openshift/enhancements/blob/d09fbc431dcab82e730641900d5c97571b992153/enhancements/ingress/default-ingress-cert-configmap.md>), which is to say there is a bit more work to do in the authentication operator besides turning off the route check.

Reply:
> By the way, the authentication operator also expects the ingress operator to create the "router-certs" secret and the "default-ingress-cert" configmap with the default ingress wildcard certificate, which is to say there is a bit more work to do in the authentication operator besides turning off the route check.

Yes, there are quite some dependencies between the authentication and ingress operators. However they are not relevant for this EP, as the authentication server is not managed by the authentication operator on HyperShift. So, the authentication operator is not used on HyperShift.

Reply:
> Yes, there are quite some dependencies between the authentication and ingress operators. However they are not relevant for this EP, as the authentication server is not managed by the authentication operator on HyperShift. So, the authentication operator is not used on HyperShift.

This section is specifically about what would be required for standalone OpenShift though, so these details do seem relevant here.  I'm fine with some hand-waving though (that is, to do W, we would need to do X, Y, and Z, and (wave hands) maybe some more things, but X, Y, and Z are reason enough not to do W).

Reply:
The links to the usage of the ingress certificates added.